src/app/contacts/services/contact.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@angular/core';
+import { Contact } from '../models/contactDTO';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { BehaviorSubject } from 'rxjs';
+import { environment } from '../../../environments/environment';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class ContactService {
+  contacts: Contact[];
+  contactsSource = new BehaviorSubject([]);
+
+  constructor(private http: HttpClient) {
+    this.getContacts()
+  }
+
+  getContacts = () => {
+    let headers = new HttpHeaders();
+    headers = headers.append('Authorization', 'Basic ' + btoa('MY_SECRET_ENCRYPTED_TOKEN'));
+    this.http.get<Contact[]>(environment.baseUrl + 'contacts.json', { headers: headers }).subscribe(contacts => {
+      this.contacts = contacts.sort((one, two) => (one.id > two.id ? -1 : 1));
+      this.contactsSource.next(this.contacts);
+    });
+  }
+  deleteContacts = (contacts: Contact[]) => {
```

Comment:
It was supposed to use normal API calls, but the API server is mocked (ex: json-server)

Reply:
need clarification please